tests/foreman/ui/test_host.py

```diff
@@ -1701,3 +1704,194 @@ def test_positive_delete_libvirt(
+
+
+@pytest.fixture
+def class_gce_hostgroup(
+        module_org,
+        module_loc,
+        default_partition_table,
+        default_architecture,
+        module_os,
+        module_environment,
+        module_proxy,
+        class_gce_domain,
+        class_gce_resource_with_image,
+        module_lce,
+        module_content_view,
+):
+    return entities.HostGroup(
+        architecture=default_architecture,
+        compute_resource=class_gce_resource_with_image,
+        domain=class_gce_domain,
+        lifecycle_environment=module_lce,
+        content_view=module_content_view,
+        location=[module_loc],
+        environment=module_environment,
+        puppet_proxy=module_proxy,
+        puppet_ca_proxy=module_proxy,
+        content_source=module_proxy,
+        operatingsystem=module_os,
+        organization=[module_org],
+        ptable=default_partition_table,
+    ).create()
+
+
+@tier4
+@skip_if_not_set('gce')
+def test_positive_gce_provision_end_to_end(
+        session,
+        module_org,
+        module_loc,
+        module_os,
+        class_gce_domain,
+        class_gce_hostgroup,
+        gce_client
+):
+    """Provision Host on GCE compute resource
+
+    :id: 8d1877bb-fbc2-4969-a13e-e95e4df4f4cd
+
+    :expectedresults: Host is provisioned successfully
+
+    :CaseLevel: System
+    """
+    name = gen_string('alpha').lower()
+    hostname = u'{0}.{1}'.format(name, class_gce_domain.name)
+    gceapi_vmname = hostname.replace('.', '-')
+    root_pwd = gen_string('alpha', 15)
+    with Session('gce_tests') as session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        # Provision GCE Host
+        try:
+            session.host.create({
+                'host.name': name,
+                'host.hostgroup': class_gce_hostgroup.name,
+                'provider_content.virtual_machine.machine_type': 'g1-small',
+                'provider_content.virtual_machine.external_ip': True,
+                'provider_content.virtual_machine.network': 'default',
+                'operating_system.operating_system': module_os.title,
+                'operating_system.image': 'autogce_img',
+                'operating_system.root_password': root_pwd
+            })
+            wait_for(
+                lambda: session.host.get_details(hostname)[
+                            'properties']['properties_table']['Build'] != 'Pending installation',
+                timeout=600,
+                delay=30,
+                fail_func=session.browser.refresh,
+                silent_failure=True,
+                handle_exception=True,
+            )
+            # 1. Host Creation Assertions
+            # 1.1 UI based Assertions
+            host_info = session.host.get_details(hostname)
+            assert session.host.search(hostname)[0]['Name'] == hostname
+            assert host_info['properties']['properties_table']['Build'] == 'Installed'
```

Comment:
Can we add assertion here to see `Zone`  `IP`, 'machineType', `Interface` as well , as that would help to make sure there is not empty values around it ?

Reply:
I am already testing IP, Zone and machineType already and that too directly using GCE APIs they are below this assert statement.

What do you want to check in interface, I could only see these fields available in GCE Host:
![Screenshot from 2019-08-14 20-04-36](https://user-images.githubusercontent.com/11752425/63029799-cf2ec500-bece-11e9-8890-667e8a21f90a.png)

Reply:
the idea is to check the values around these field are as expected on Satellite ..

As you are doing validation of GCE directly hence does not shows the validation at Satellite as well for same VM/host.

About interface please ignore as nothing much exists .

Reply:
> As you are doing validation of GCE directly(that is +1 as validating there too) hence does not show the validation at Satellite as well for same VM/host.

didnt get this, but Do you sill want me to test things from UI ?

Reply:
The thought it :

If you are checking `IP, Zone and machineType` using GCE API then can you please check the same at **Satellite end as well** ?
Why I am saying to check it because it may happen that sometime related values(IP, Zone and machineType) are not populating at Satellite end but from GCE IP you able to see it.

I believe `host_info` would able to provide these details and you only need to add the related **assert** .

Reply:
@vijay8451 Only IP value is accessible in host_info. For getting Zone and machineType I have to switch to the VM tab in the host page for which support/PR needs to be airgun.

Currently I am trying so, but unfortunately, the VM tab doesn't have an ID or something unique to add here: https://github.com/SatelliteQE/airgun/blob/master/airgun/views/host.py#L491

I am looking into it.

Reply:
@jyejare I have another thought , let's validate IP here only on UI...
and keep  `Zone and machineType` validation through API(whenever GCE API support added to Satellite) ... what do you think ?

Reply:
@vijay8451 , Ok then! I will straightaway update the PR, that would be quick.